src/main/java/randoop/resource/generator/MethodListGen.java

```diff
@@ -0,0 +1,238 @@
+package randoop.resource.generator;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static randoop.reflection.OperationModel.signatureToOperation;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.plumelib.reflection.Signatures;
+import randoop.main.RandoopUsageError;
+import randoop.reflection.EverythingAllowedPredicate;
+import randoop.reflection.VisibilityPredicate;
+import scenelib.annotations.Annotation;
+import scenelib.annotations.el.AClass;
+import scenelib.annotations.el.AMethod;
+import scenelib.annotations.el.AScene;
+import scenelib.annotations.io.classfile.ClassFileReader;
+
+/**
+ * This tool generates the resource files JDK-sef-methods.txt and omitmethods-defaults.txt, which
+ * contain a list of side effect free methods and a list of methods to omit during Randoop
+ * execution, respectively.
+ */
+public class MethodListGen {
+  // The extension for a Java .class file. These files contain annotations that
+  // this tool will parse.
+  private static final String CLASS_EXT = ".class";
+
+  /** The type annotations indicating a non-deterministic return value. */
+  private static final Collection<String> NONDET_ANNOTATIONS =
+      Collections.singletonList("org.checkerframework.checker.determinism.qual.NonDet");
+  /** The method annotations indicating a side-effect-free method. */
+  private static final Collection<String> SEF_ANNOTATIONS =
+      Arrays.asList(
+          "org.checkerframework.dataflow.qual.SideEffectFree",
+          "org.checkerframework.dataflow.qual.Pure");
+
+  /** Text to place at the beginning of file {@code omitmethods-defaults.txt}. */
+  private static final String OMIT_METHODS_FILE_HEADER =
+      "# Long-running.  With sufficiently small arguments, can be fast.\n"
```

Comment:
Never use `\n` in source code.  A program should respect the newline convention of the host operating system.  Use `System.lineSeparator()` (possibly together with String.join).

Reply:
Thanks! I will definitely try to remember that.